server/src/main/kotlin/suwayomi/tachidesk/global/impl/sync/SyncManager.kt

```diff
@@ -0,0 +1,406 @@
+                            }
+                        }
+
+                        // Handle non-favorites
+                        !remoteManga.favorite -> {
+                            logger.debug { "Adding to non-favorites: ${remoteManga.title}" }
+                            nonFavorites.add(remoteManga)
+                        }
+                    }
+                }
+            }
+
+        logger.debug {
+            "Filtering completed in $elapsedTime. Favorites found: ${favorites.size}, Non-favorites found: ${nonFavorites.size}"
+        }
+
+        return Pair(favorites, nonFavorites)
+    }
+
+    private fun updateNonFavorites(nonFavorites: List<BackupManga>) {
+        nonFavorites.forEach { nonFavorite ->
+            val localManga =
+                transaction {
+                    MangaTable
+                        .selectAll()
+                        .where {
+                            (MangaTable.sourceReference eq nonFavorite.source) and
+                                (MangaTable.url eq nonFavorite.url)
+                        }.limit(1)
+                        .map { MangaTable.toDataClass(it) }
+                        .firstOrNull()
+                }
+
+            if (localManga != null) {
+                if (localManga.inLibrary != nonFavorite.favorite) {
+                    val updatedManga = localManga.copy(inLibrary = nonFavorite.favorite)
+                    updateManga(updatedManga)
+                }
+            }
+        }
+    }
+
+    private fun updateManga(manga: MangaDataClass) {
+        transaction {
+            MangaTable.update({ MangaTable.id eq manga.id }) {
+                it[MangaTable.url] = manga.url
+                it[MangaTable.title] = manga.title
+                it[MangaTable.initialized] = manga.initialized
+
+                it[MangaTable.artist] = manga.artist
+                it[MangaTable.author] = manga.author
+                it[MangaTable.description] = manga.description
+                it[MangaTable.genre] = manga.genre.joinToString(separator = ", ")
+
+                it[MangaTable.status] = MangaStatus.valueOf(manga.status).value
+                it[MangaTable.thumbnail_url] = manga.thumbnailUrl
+                it[MangaTable.thumbnailUrlLastFetched] = manga.thumbnailUrlLastFetched
+
+                it[MangaTable.inLibrary] = manga.inLibrary
+                it[MangaTable.inLibraryAt] = manga.inLibraryAt
+
+                it[MangaTable.sourceReference] = manga.sourceId.toLong()
+
+                it[MangaTable.realUrl] = manga.realUrl
+                it[MangaTable.lastFetchedAt] = manga.lastFetchedAt ?: 0L
+                it[MangaTable.chaptersLastFetchedAt] = manga.chaptersLastFetchedAt ?: 0L
+
+                it[MangaTable.updateStrategy] = manga.updateStrategy.name
+
+                it[MangaTable.version] = manga.version
+            }
+        }
+    }
```

Comment:
Seems like we can simplify this. Also we could handle the thumbnail here.
```suggestion
                    transaction {
                        MangaTable.update({ MangaTable.id eq localManga.id }) {
                            it[inLibrary] = nonFavorite.favorite
                        }
                    }.apply {
                        handleMangaThumbnail(localManga.id, nonFavorite.favorite)
                    }
                }
            }
        }
    }
```